File: NMSMB Scripts/Jackty89/CheapPetSlots.cs
//=============================================================================
//Author: Jackty89

//=============================================================================

namespace cmk.NMS.Scripts.Mod
{
	public class CheapPetSlots : cmk.NMS.ModScript
	{
		protected override void Execute()
		{
			GcDebugOptions();
		}

		//...........................................................

		protected void GcDebugOptions()
		{
			var mbin = ExtractMbin<GcCostTable>("METADATA\\REALITY\\TABLES\\COSTTABLE.MBIN");
			GcCostMoneyList petSlotCost = (GcCostMoneyList)mbin.SimpleInteractionTable.Find(ENTRY => ENTRY.Id == "C_PET_SLOT").Cost;
			petSlotCost.Costs[0] = 25;
			petSlotCost.Costs[1] = 50;
			petSlotCost.Costs[2] = 75;
			petSlotCost.Costs[3] = 100;
			petSlotCost.Costs[4] = 125;
			petSlotCost.Costs[5] = 150;
		}
	}
}

//=============================================================================

File: Jackty89/CheapPetSlots.lua
ModName = "CheapPetSlots"
Author = "Jackty89"

CostTablePath = "METADATA/REALITY/TABLES/COSTTABLE.MBIN"
Cost_List = [[
    <Property name="Costs" value="25" _index="0" />
    <Property name="Costs" value="50" _index="1" />
    <Property name="Costs" value="75" _index="2" />
    <Property name="Costs" value="100" _index="3" />
    <Property name="Costs" value="125" _index="4" />
    <Property name="Costs" value="150" _index="5" />
]]

NMS_MOD_DEFINITION_CONTAINER =
{
    MOD_FILENAME = ModName,
    MOD_DESCRIPTION = ModName,
    MOD_AUTHOR = Author,
    MODIFICATIONS =
    {
        {
            MBIN_CHANGE_TABLE =
            {
                {
                    MBIN_FILE_SOURCE = CostTablePath,
                    MXML_CHANGE_TABLE =
                    {
                        {
                            SPECIAL_KEY_WORDS = {"Id", "C_PET_SLOT"},
                            PRECEDING_KEY_WORDS = {"Costs"},
                            CREATE_HOES = "TRUE",
                        },
                        {
                            SPECIAL_KEY_WORDS = {"Id", "C_PET_SLOT"},
                            PRECEDING_KEY_WORDS = {'Costs'},
                            CREATE_HOS = "TRUE",
                            ADD = Cost_List
                        }
                    }
                }
            }
        }
    }
}